llvm/lib/CodeGen/CriticalAntiDepBreaker.cpp

```diff
@@ -66,10 +66,10 @@ void CriticalAntiDepBreaker::StartBlock(MachineBasicBlock *BB) {
   // Examine the live-in regs of all successors.
   for (const MachineBasicBlock *Succ : BB->successors())
     for (const auto &LI : Succ->liveins()) {
-      for (MCRegAliasIterator AI(LI.PhysReg, TRI, true); AI.isValid(); ++AI) {
-        MCRegister Reg = *AI;
-        Classes[Reg.id()] = reinterpret_cast<TargetRegisterClass *>(-1);
-        KillIndices[Reg.id()] = BBSize;
+      for (MCRegAliasIterator AI(LI, TRI, true); AI.isValid(); ++AI) {
+        unsigned Reg = (*AI).id();
+        Classes[Reg] = reinterpret_cast<TargetRegisterClass *>(-1);
+        KillIndices[Reg] = BBSize;
```

Comment:
Don't change the `Reg.id()` part. It just makes the patch bigger and harder to review.